package/network/utils/comgt/files/ncm.sh

```diff
@@ -117,21 +127,23 @@ proto_ncm_setup() {
 	}
 
 	echo "Starting network $interface"
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
-
 	echo "Setting up $ifname"
-	
+
 	proto_init_update "$ifname" 1
 	proto_add_data
 	json_add_string "manufacturer" "$manufacturer"
 	proto_close_data
 	proto_send_update "$interface"
 
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_select "$manufacturer"
+	json_get_vars connect
+	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+		echo "Failed to connect"
+		proto_notify_error "$interface" CONNECT_FAILED
+		return 1
+	}
+
```

Comment:
All right, keep the configure block then. For consistency's sake move that command for other modems as well in the ncm.json file. Also move the lines `"AT$QCPDPP=${profile},${auth:-0}${password:+,\\\"$password\\\"}${username:+,\\\"$username\\\"}"` for Sierra and `"AT*EIAAUW=${profile},1,\\\"${username}\\\",\\\"${password}\\\",${auth:-00111}"` for Sony to the new configure block.